core/lib/dal/src/blocks_dal.rs

```diff
@@ -756,6 +939,52 @@ impl BlocksDal<'_, '_> {
         Ok(())
     }
 
+    pub async fn get_ready_for_precommit_txs(
+        &mut self,
+        l1_batch_number: Option<L1BatchNumber>,
+    ) -> DalResult<Vec<TxForPrecommit>> {
+        let mut tx = self.storage.start_transaction().await?;
+        // Miniblocks belongs to the non sealed batches don't have batch number,
+        // so for the pending batch we use None
+
+        let txs = sqlx::query!(
+            r#"
+            SELECT
+                transactions.hash, transactions.error,
+                miniblock_number AS "miniblock_number!",
+                miniblocks.timestamp
+            FROM transactions
+            JOIN miniblocks ON transactions.miniblock_number = miniblocks.number
+            WHERE
+                (
+                    miniblocks.l1_batch_number IS NULL AND $2
+                    OR miniblocks.l1_batch_number = $1
+                )
+                AND
+                miniblocks.rolling_txs_hash IS NOT NULL
+                AND
+                miniblocks.eth_precommit_tx_id IS NULL
```

Comment:
ok, looks like in this place the index I was talking about in the first comment is also required, so maybe it's worth just adding it. Its impact on miniblock header insertion time most likely will be negligible.
Also I think it makes sense to invert JOIN, i.e. select from miniblocks and join transactions, this way it will be guaranteed that index will be used.